paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/FlinkCatalog.java

```diff
@@ -212,8 +222,16 @@ public boolean tableExists(ObjectPath tablePath) throws CatalogException {
     @Override
     public void dropTable(ObjectPath tablePath, boolean ignoreIfNotExists)
             throws TableNotExistException, CatalogException {
+        Identifier identifier = toIdentifier(tablePath);
+        Table table = null;
         try {
+            if (catalog.tableExists(identifier)) {
```

Comment:
This brings some overhead for normal tables.
Can we just have a
`if we need to unregister -> a separate branch`.

Reply:
Currently we create log store topic for each table, we cannot check whether a table needs to unregister logsystem before we get its options. What do you think of  it?